test_create_bitfield failing on s390x

Attempting to update the Fedora h5py package to 3.16.0 we are getting the following failure:
```
=================================== FAILURES ===================================
_____________________________ test_create_bitfield _____________________________
writable_file = <HDF5 file "test.h5" (mode r+)>
    def test_create_bitfield(writable_file):
        data = np.arange(12, dtype=np.uint16)
        ds = writable_file.create_dataset(
            make_name("x"), dtype=h5py.Datatype(h5py.h5t.STD_B16LE), shape=(12,)
        )
>       ds[:] = data
        ^^^^^
serial/build/lib.linux-s390x-cpython-314/h5py/tests/test_dtype.py:540: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
serial/build/lib.linux-s390x-cpython-314/h5py/_hl/dataset.py:1110: in __setitem__
    self.id.write(mspace, fspace, val, mtype, dxpl=self._dxpl)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
h5py/h5d.pyx:305: in h5py.h5d.DatasetID.write
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ???
E   OSError: Can't synchronously write data (no appropriate function for conversion path)
h5py/_proxy.pyx:122: OSError
```
Bull build log here: https://kojipkgs.fedoraproject.org//work/tasks/4517/143144517/build.log
This is with hdf5 1.14.6
Builds fine on other arches.

--- a/h5py/tests/test_dtype.py
+++ b/h5py/tests/test_dtype.py
@@ -533,7 +533,7 @@ def test_opaque(writable_file):
 
 
 def test_create_bitfield(writable_file):
-    data = np.arange(12, dtype=np.uint16)
+    data = np.arange(12, dtype="<u2")
     ds = writable_file.create_dataset(
         make_name("x"), dtype=h5py.Datatype(h5py.h5t.STD_B16LE), shape=(12,)
     )

